src/components/LocationAutocompleteInput/LocationAutocompleteInputImpl.js

```diff
@@ -161,6 +172,32 @@ class LocationAutocompleteInputImplementation extends Component {
 
   componentDidMount() {
     this._isMounted = true;
+
+    // Check whether the new version of the Google Places API is in use.
+    // The useNewGooglePlacesAPI property indicates whether the updated
+    // Google Places API is available and being utilized. If the useNewGooglePlacesAPI
+    // property is undefined, it means we haven't determined the API version yet.
+    if (typeof window !== 'undefined') {
```

Comment:
This is unnecessary!

The **componentDidMount** is executed after the component has mounted to DOM (aka it will have window since this function is run on browser side)